### PR TITLE
Fixed #31069, Refs #26431 -- Doc'd RegexPattern behavior change in passing optional named groups in Django 3.0.

### PR DESCRIPTION
**Pull request by hramezani**
[Ticket 31069](https://code.djangoproject.com/ticket/31069)

### CODE REVIEW

File: docs/topics/http/urls.txt [hunk 1]
```diff
@@ -58,7 +58,12 @@ algorithm the system follows to determine which Python code to execute:
    * The keyword arguments are made up of any named parts matched by the
      path expression, overridden by any arguments specified in the optional
      ``kwargs`` argument to :func:`django.urls.path` or
-     :func:`django.urls.re_path`.
+     :func:`django.urls.re_path`. Django don't pass anything for named groups
+     that are not provided.
```
Comment (felixxm) [inline docs/topics/http/urls.txt]: Maybe:
```
    * The keyword arguments are made up of any named parts matched by the
      path expression that are provided, overridden by any arguments specified
      in the optional ``kwargs`` argument to :func:`django.urls.path` or
      :func:`django.urls.re_path`.

File: docs/topics/http/urls.txt [hunk 2]
```diff
@@ -58,7 +58,12 @@ algorithm the system follows to determine which Python code to execute:
...
+
+     .. versionchanged:: 3.0
+
+          Django don't pass ``None`` for named groups that are not provided.
```
Comment (felixxm) [inline docs/topics/http/urls.txt]: Wording suggestion:
```
.. versionchanged:: 3.0 
    In older versions, the keyword arguments with ``None`` values are made up
    also for not provided named parts.

**Comment by hramezani** [comment]
@felixxm Thanks for your review and suggestion.
what do you think about the following release not for `Backwards incompatible changes`:

```
*‌ The keyword arguments with ``None`` values in matched URL pattern are no
  longer get passed to view, also for not provided named parts.
```

**Comment by felixxm** [comment]
We can take a `releases/3.0.1.txt` as a pattern :thinking: , e.g.:
```
* ``RegexPattern``, used by :func:`~django.urls.re_path`, no longer return a
  keyword argument with ``None`` value to be passed to the view for the
  optional named group that is missing.
```

**Comment by hramezani** [comment]
> * ``RegexPattern``, used by :func:`~django.urls.re_path`, no longer return a
>   keyword argument with ``None`` value to be passed to the view for the
>   optional named group that is missing.

Thanks, I just replaced `no longer return` with `no longer returns`.

File: docs/topics/http/urls.txt [hunk 3]
```diff
@@ -56,9 +56,14 @@ algorithm the system follows to determine which Python code to execute:
...
-     ``kwargs`` argument to :func:`django.urls.path` or
-     :func:`django.urls.re_path`.
+     path expression that are provided, overridden by any arguments specified
+     in the optional ``kwargs`` argument to :func:`django.urls.path` or
+     :func:`django.urls.re_path`. .
+
+     .. versionchanged:: 3.0
+
+     In older versions, the keyword arguments with ``None`` values are made up
+     also for not provided named parts.
```
Comment (felixxm) [inline docs/topics/http/urls.txt]:
```
    .. versionchanged:: 3.0

          In older versions...

Comment (hramezani) [reply docs/topics/http/urls.txt]: Sorry. fixed.

**Comment by felixxm** [comment]
@hramezani Thanks :+1:

**Pull request merged**